likecoin/js/sidebar/src/pages/LikeCoinPlugin.js

```diff
@@ -33,135 +29,108 @@ function LikeCoinPlugin(props) {
   const [mattersArticleSlug, setMattersArticleSlug] = useState(props.DBMattersArticleSlug);
   const [fingerprints, setFingerprints] = useState([]);
   const [shouldStartProcess, setShouldStartProcess] = useState(false);
-  const [shouldUploadArweave, setShouldUploadArweave] = useState(false);
-  const [shouldStartLIKEPay, setShouldStartLIKEPay] = useState(false);
-  const [shouldSendISCNReadyMessage, setShouldSendISCNReadyMessage] = useState(false);
-  const [shouldSkipArweave, setShouldSkipArweave] = useState(false);
+  const [shouldSendISCNRegisterData, setShouldSendISCNRegisterData] = useState(false);
   const [popUpWindow, setPopUpWindow] = useState(null);
-  const uploadToArweave = useCallback(
-    async (data) => {
-      try {
-        const { tx_hash: txHash } = data;
-        props.postArweaveUploadAndIPFSData({ txHash });
-      } catch (error) {
-        console.error('Error occurs when uploading to Arweave:');
-        console.error(error);
-      }
+  const onArweaveCallback = useCallback(
+    (data) => {
+      const {
+        ipfsHash, arweaveId: newArweaveId,
+      } = data;
+      props.postArweaveInfoData({
+        ipfsHash, arweaveId: newArweaveId,
+      });
     },
     [props],
   );
-  const onLIKEPayCallback = useCallback(
-    async (event) => {
-      if (event && event.data && event.origin === 'https://like.co' && typeof event.data === 'string') {
-        const { action, data } = JSON.parse(event.data);
-        if (action === 'TX_SUBMITTED') {
-          try {
-            await uploadToArweave(data);
-          } catch (error) {
-            console.error(error);
-          }
-        }
-      }
+  const onISCNCallback = useCallback(
+    (data) => {
+      const {
+        tx_hash: txHash, iscnId, iscnVersion, timestamp,
+      } = data;
+      props.postISCNInfoData({
+        iscnHash: txHash,
+        iscnId,
+        iscnVersion,
+        timestamp,
+      });
     },
-    [uploadToArweave],
+    [props],
```

Comment:
useCallback will not be deeply compared props, is it a concern?

Reply:
`props.postISCNInfoData` won't change, but is there a better style to write this?